packages/uploadthing/src/internal/handler.ts

```diff
@@ -19,16 +22,35 @@ import { UPLOADTHING_VERSION } from "../constants";
 import { conditionalDevServer } from "./dev-hook";
 import { getParseFn } from "./parser";
 import { VALID_ACTION_TYPES } from "./types";
-import type { ActionType, FileRouter } from "./types";
+import type { ActionType, FileRouter, UTEvents } from "./types";
+
+/**
+ * Creates a wrapped fetch that will always forward a few headers to the server.
+ */
+const createUTFetch = (apiKey: string) => {
+  return async (endpoint: `/${string}`, payload: unknown) => {
```

Comment:
Potentially needs a method override if we use it everywhere. eg. polling is a `GET` iirc

Reply:
yup - I'm thinking of expanding this more later with generated types from infra side so will tackle that then, this is enough for now